apps/server/src/modules/system/service/system.service.spec.ts

```diff
@@ -33,54 +68,274 @@ describe('SystemService', () => {
 		orm = await setupEntities();
 		systemRepo = module.get(SystemRepo);
 		systemService = module.get(SystemService);
-	});
 
-	beforeEach(() => {
-		oauthSystems = [];
-		const iserv = systemFactory.buildWithId();
-		oauthSystems.push(iserv);
+		oidcSystems.push(oidc1);
+		oidcSystems.push(oidc2);
+
 		allSystems.push(iserv);
-		allSystems.push(systemFactory.buildWithId({ oauthConfig: undefined }));
-		const moodle = systemFactory.buildWithId();
-		moodle.type = 'moodle';
-		allSystems.push(moodle);
+		allSystems.push(keycloak);
+		allSystems.push(...oidcSystems);
+	});
 
-		systemRepo.findByFilter.mockResolvedValue(oauthSystems);
+	beforeEach(() => {
 		systemRepo.findAll.mockResolvedValue(allSystems);
```

Comment:
Optional change:
All of these mock initialization could be moved to `beforeAll()`. If the mock implementation is defined during the creation of the mock you can reset it to the default mock implementation with `mockReset()`.
If the mock needs to be changed in only few tests, `.mockImplementationOnce()` can help as well to avoid redefining the mock all the time.

Reply:
done

Reply:
I see you managed to change it. Looks like `createMock(...)` does not set proper default mock implementations.
My guess was that something like the following would work, but apparently not. On my machine currently tests are broken. I can't try.
```createMock({
findAll: () => allSystems,
...
})
```